src/astro_image_display_api/dummy_viewer.py

```diff
@@ -99,6 +99,15 @@ def load_fits(self, file: str | os.PathLike) -> None:
             The FITS file to load. If a string, it can be a URL or a
             file path.
         """
+        if isinstance(file, (str, os.PathLike)):
+            self._load_fits(file)
```

Comment:
Try to use `_load_asdf` here?
```suggestion
        if isinstance(file, (str, os.PathLike)):
            if isinstance(file, str):
                is_adsf = file.endswith(".asdf")
            else:
                is_asdf = file.suffix == ".asdf"
            if is_asdf:
                self._load_asdf(file)
            else:
                self._load_fits(file)
```